### PR TITLE
gh-100931: Test all `pickle` protocols in `test_slice`

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-100931

### CODE REVIEW

File: Lib/test/test_json/test_attrdict.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def test_pickle(self):
         cached_module = sys.modules.get('json')
         sys.modules['json'] = self.json
         try:
-            for protocol in range(6):
+            for protocol in range(pickle.HIGHEST_PROTOCOL + 1):
```
Comment (sobolevn) [inline Lib/test/test_json/test_attrdict.py:136]: This change is not related, but I think it is minor enough to be included. It just replaces a magic number `6` with more context.

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Comment by AlexWaygood** [comment]
@sobolevn, can you sort the backports? :)

**Comment by sobolevn** [comment]
Yes, will do a bit later.

**Pull request merged**